src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -1936,7 +1936,34 @@ private Type RetrieveClrTypeForConstant(IEdmTypeReference edmTypeReference, ref
                 Contract.Assert(strValue != null);
 
                 constantType = Nullable.GetUnderlyingType(constantType) ?? constantType;
-                value = Enum.Parse(constantType, strValue);
+
+                IEdmEnumType enumType = edmTypeReference.AsEnum().EnumDefinition();
+                ClrEnumMemberAnnotation memberMapAnnotation = Model.GetClrEnumMemberAnnotation(enumType);
+                if (memberMapAnnotation != null)
+                {
+                    IEdmEnumMember enumMember = enumType.Members.FirstOrDefault(m => m.Name == strValue);
+                    if (enumMember == null)
+                    {
+                        enumMember = enumType.Members.FirstOrDefault(m => m.Value.ToString() == strValue);
+                    }
```

Comment:
is this code path tested?

Reply:
Actually, yes. If I don't add this, there's one test case failing because the ODataEnumValue has the "1" as the value, not the string. But, i can't remember the test (name).
If that matters, i can figure it out later

---
In reply to: [485247028](https://github.com/OData/WebApi/pull/2267#discussion_r485247028) [](ancestors = 485247028)